python/lsst/ts/ofc/scripts/generate_normalization_weights.py

```diff
@@ -0,0 +1,193 @@
+        fwhm_weights[i] = np.sqrt(np.sum(np.square(fwhm_matrix[:, i])))
+
+    normalization_weights = range_weights * fwhm_weights
+
+    return normalization_weights
+
+
+def save_normalization_weights(
+    normalization_weights: np.ndarray, output_dir: str
+) -> None:
+    """Save the normalization weights.
+
+    Parameters
+    ----------
+    normalization_weights : np.ndarray
+        Normalization weights.
+    output_dir : str
+        Output directory to save the normalization weights.
+    """
+    # Complete path to the yaml file
+    yaml_file_path = f"{output_dir}/normalization_weights.yaml"
+
+    # Write header and normalization weights to the yaml file
+    with open(yaml_file_path, "w") as yaml_file:
+        header = (
+            "---\n"
+            "# Normalization weights for the degrees of freedom\n"
+            "# Description: Normalization weights for the sensitivity matrix\n"
+            "#\n"
+            "# The normalization weights are used to normalize\n"
+            "# the sensitivity matrix and be able to operate in\n"
+            "# a different basis. The normalization weights are\n"
+            "# effectively a rescaling diagonal matix that is\n"
+            "# applied to the sensitivity to equalize the relevance\n"
+            "# of the different degrees of freedom (namelly, the\n"
+            "# different units and ranges of the degrees of freedom).\n"
+            "# The default is simply an identity (weights = 1)),\n"
+            "# which means that the sensitivity matrix is not\n"
+            "# normalized in any way. Units can be dimensionless.\n"
+            "# The reduced basis weights units are um of range * arcsec of FWHM.\n"
+            "# But the matrix is denormalized before retrieving the optical state,\n"
+            "# so it should not be relevant.\n\n"
+        )
+        yaml_file.write(header)
+        yaml.dump(normalization_weights.tolist(), yaml_file)
+
+
+def main(args: argparse.Namespace) -> None:
+    """Main function to generate normalization weights.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Arguments parsed from the command-line.
+    """
+    # Load the OFC data
+    ofc_data = OFCData(args.instrument)
+
+    # Define the sensors to use
+    sensor_name_list = ["R00_SW0", "R04_SW0", "R40_SW0", "R44_SW0"]
+
+    # Get the field angles for the sensors
+    field_angles = [ofc_data.sample_points[sensor] for sensor in sensor_name_list]
+
+    # Load the sensitivity matrix
+    dz_sensitivity_matrix = SensitivityMatrix(ofc_data)
+
+    # Compute the FWHM matrix
+    fwhm_matrix = compute_fwhm_matrix(ofc_data, dz_sensitivity_matrix, field_angles)
+
+    # Compute the normalization weights
+    normalization_weights = compute_normalization_weights(ofc_data, fwhm_matrix)
+
+    # Save the normalization weights
+    save_normalization_weights(normalization_weights, args.output_dir)
+
+
+def generate_normalization_weights() -> None:
```

Comment:
This method has only one line and it only does:

```
main(parse_arguments())
```

Please, either remove it or move the code from `main` to here and remove `main`.